### PR TITLE
Get user whitelist to work again

### PR DESCRIPTION
**Pull request by Niharika29**
This commit:
* Removes Redis in favor of API call to get whitelist
* Handle error when whitelist page is empty

### CODE REVIEW

File: src/Controllers/CopyPatrol.php [hunk 1]
```diff
@@ -108,7 +108,7 @@ protected function handleGet() {
 			return $this->render( 'index.html' );
 		}
 		$userWhitelist = [];
```
Comment (MusikAnimal) [inline src/Controllers/CopyPatrol.php]: don't need this anymore

File: src/Controllers/CopyPatrol.php [hunk 2]
```diff
@@ -298,18 +298,8 @@ private function getUserWhitelist() {
...
-			$whitelist = $cacheItem->get( $cacheKey );
-		} else {
-			// It doesn't exist or it expired, so fetch from wiki page.
-			$whitelist = $this->wikiDao->getUserWhitelist();
-			// Store in the cache for 2 hours.
-			$cacheItem->set( $whitelist )->expiresAfter( 2 * 60 * 60 );
-			$this->cache->save( $cacheItem );
-		}
+		// It doesn't exist so fetch from wiki page.
+		$whitelist = $this->wikiDao->getUserWhitelist();
```
Comment (MusikAnimal) [inline src/Controllers/CopyPatrol.php]: Let's try using the filesystem cache locally (remove redis vars from .env), and if that works for everyone maybe we could keep the caching functionality.

Comment (Niharika29) [inline src/Controllers/CopyPatrol.php]: Yep, that worked fine.

File: src/Controllers/CopyPatrol.php (hunk 1 above)
Comment (Niharika29) [inline src/Controllers/CopyPatrol.php]: Removed.

**Comment by MusikAnimal** [comment]
Looks good to me, but let's get #42 merged first since it's including in this. See my comment there

File: src/Controllers/CopyPatrol.php [hunk 3]
```diff
@@ -288,7 +287,7 @@ protected function getFilterTypes() {
 	}
 
 	/**
-	 * Get cached user watchlist or re-fetch from wiki page and update redis
+	 * Get user watchlist from wiki page
```
Comment (MusikAnimal) [inline src/Controllers/CopyPatrol.php]: restore original comment

File: src/Controllers/CopyPatrol.php [hunk 4]
```diff
@@ -298,8 +297,7 @@ private function getUserWhitelist() {
 		if ( $whitelist !== null ) {
 			return $whitelist;
 		}
-		// Get whitelist from the cache if possible.
```
Comment (MusikAnimal) [inline src/Controllers/CopyPatrol.php]: and maybe keep this comment, too

**Comment by MusikAnimal** [comment]
Still confused why [this](https://github.com/wikimedia/CopyPatrol/pull/42/files#diff-d3bb9db39e409dbe4b7f9ce9de064a71R92) is showing up given it's already in master, but whatever, the code is there, so :+1: from me!

**Comment by Niharika29** [comment]
Gonna squash commits. This is messy.

**Comment by MusikAnimal** [comment]
One thing I didn't do was make sure this works when `Utilisateur:Example` is listed, as opposed to `User:Example`. I think it will

**Comment by Niharika29** [comment]
I tested it. Works fine.

**Pull request merged**